### PR TITLE
Add provider configs

### PR DESCRIPTION
**Pull request by ZuGao**
Resolves https://github.com/projectsyn/component-crossplane/issues/33

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 1]
```diff
@@ -130,6 +130,27 @@ The component doesn't validate the provided ControllerConfig spec.
...
+type:: dict
+default:: `{}`
+example::
++
+[source,yaml]
+----
+"provider-name-two":
+  spec:
+  credentials:
+    source: InjectedIdentity
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/parameters.adoc:145]: Indentation error, and the `apiVersion` is a required field.
```suggestion
  apiVersion: helm.crossplane.io/v1beta1
  spec:
    credentials:
      source: InjectedIdentity
```

File: tests/golden/defaults-with-provider-configs/crossplane/crossplane/70_provider_configs.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    argocd.argoproj.io/sync-wave: '10'
+  labels:
+    name: provider-name-one
+  name: provider-name-one
+spec:
+  credentials:
+    source: InjectedIdentity
+---
+apiVersion: pkg.crossplane.io/v1
+kind: ProviderConfig
```
Comment (ccremer) [inline tests/golden/defaults-with-provider-configs/crossplane/crossplane/70_provider_configs.yaml:15]: There is no `ProviderConfig` kind in the `pkg.crossplane.io` group. (https://doc.crds.dev/github.com/crossplane/crossplane)
Can we make the field `apiVersion` required so it doesn't default to `pkg.crossplane.io`?

**Review by ccremer — CHANGES_REQUESTED** [review]

File: tests/golden/defaults-with-provider-configs/crossplane/crossplane/70_provider_configs.yaml (hunk 2 above)
Comment (ZuGao) [reply tests/golden/defaults-with-provider-configs/crossplane/crossplane/70_provider_configs.yaml]: I was able to make the field mandatory, though I have no idea how to test it in golden tests.

Comment (ZuGao) [reply tests/golden/defaults-with-provider-configs/crossplane/crossplane/70_provider_configs.yaml]: The manual tests have been done successfully.

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 3]
```diff
@@ -130,6 +130,29 @@ The component doesn't validate the provided ControllerConfig spec.
...
+    credentials:
+      source: InjectedIdentity
+----
+
+This parameter allows users to deploy arbitrary provider configs.
+Each entry is transformed into a ProviderConfig resource.
+The key is used as the name of the resulting resource.
+The provided value is merged with an empty ProviderConfig resource.
+The component doesn't validate the provided ProviderConfig spec.
+ApiVersion field is mandatory.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/parameters.adoc:154]:
```suggestion
[source,yaml]
----
"provider-name-two": <1>
  apiVersion: helm.crossplane.io/v1beta1 <2>
  spec: <3>
    credentials:
      source: InjectedIdentity
----
<1> The key is used as the name of the resulting resource.
<2> Setting `apiVersion` defines in which the API group the ProviderConfig belongs to.
<3> The provided spec is merged with an empty ProviderConfig resource.

This parameter allows users to deploy arbitrary provider configs.
Each entry is transformed into a ProviderConfig resource.
The component doesn't validate the provided ProviderConfig spec.
```

**Review by ccremer — APPROVED** [review]

**Pull request merged**